python/pyarrow/_dataset.pyx

```diff
@@ -1046,6 +1046,32 @@ cdef class FileFragment(Fragment):
             self.partition_expression
         )
 
+    def slice(self, start, end):
+        """
+        Returns a new FileFragment object that will only read 
+        a slice of the old FileFragment defined by start (start byte)
+        and end (end byte).
```

Comment:
We might want to add something like "Slicing is currently only supported on CSV and the start must be at the start of a line and the end must be at the end of a line"